src/dswx_sar/pre_processing.py

```diff
@@ -0,0 +1,407 @@
+        resampled_slc or slc_demodulate: numpy.ndarray
+            numpy array of bandpassed slc
+            if resampling is True, return resampled slc with bandpass and demodulation
+            if resampling is False, return slc with bandpass and demodulation without resampling
+        meta : dict
+            dict containing meta data of bandpassed slc
+            center_frequency, rg_bandwidth, range_spacing, slant_range
+        """
+
+        self.interpolate_gdal(str(self.rtc_file_name),
+                         ancillary_file_name,
+                         os.path.join(self.scratch_dir, relocated_file_str),
+                         method)
+
+    def interpolate_gdal(self, ref_file, input_tif_str, output_tif_str, method, epsg=None):
+        print(f"> gdalwarp {input_tif_str} -> {output_tif_str}")
+
+        # get reference coordinate and projection
+        if  h5py.is_hdf5(ref_file):
+            with h5py.File(ref_file, 'r') as f4:
+                try:
+                    im = np.array(f4['/science/LSAR/GCOV/grids/frequencyA/VVVV'])
+                except:
+                    im = np.array(f4['/science/LSAR/GCOV/grids/frequencyA/HHHH'])
+                ysize, xsize = np.shape(im)
+                lon0 = np.array(f4['/science/LSAR/GCOV/grids/frequencyA/xCoordinates'])
+                lat0 = np.array(f4['/science/LSAR/GCOV/grids/frequencyA/yCoordinates'])
+                epsg_output = np.array(f4['/science/LSAR/GCOV/grids/frequencyA/projection'])
+        else:
+            reftif = gdal.Open(ref_file)
+            lat0, lon0 = self.read_x_y_array_geotiff(ref_file)
+            xsize = reftif.RasterXSize
+            ysize = reftif.RasterYSize
+            geotransform = reftif.GetGeoTransform()
+            proj = osr.SpatialReference(wkt=reftif.GetProjection())
+            epsg_output = proj.GetAttrValue('AUTHORITY',1)
+            xspacing = geotransform[1]
+            yspacing = geotransform[5]
+
+            reftif = None
+            del reftif
+
+        if ( len(lat0) != ysize ) and ( len(lon0) != xsize ):
+
+            N, S, W, E = [np.max(lat0)-yspacing/2,
+                        np.min(lat0)+yspacing/2,
+                        np.min(lon0)+xspacing/2,
+                        np.max(lon0)-yspacing/2]
+
+            print('Note: latitude shape is not same as image shape')
+
+        else:
+            N, S, W, E = [np.max(lat0), np.min(lat0), np.min(lon0), np.max(lon0)]
+        print('bounding box', N, S , W, E)
+
+        # Crop (gdalwarp)image based on geo infomation of reference image
+        if yspacing < 0: yspacing = -1 * yspacing
+
+        opt = gdal.WarpOptions(dstSRS=f'EPSG:{epsg_output}',
+                        xRes=xspacing,
+                        yRes=yspacing,
+                        outputBounds=[W,S,E,N],
+                        resampleAlg=method,
+                        format='ENVI')
+
+        ds = gdal.Warp(output_tif_str, input_tif_str, options=opt)
+        del ds
```

Comment:
```suggestion
        gdal.Warp(output_tif_str, input_tif_str, options=opt)
```
In case you are not doing anything with `ds`